include/zephyr/drivers/emul_sensor.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/emul.h>
+#include <zephyr/drivers/sensor.h>
+
```

Comment:
Please define a doxygen group similar to [this](https://github.com/zephyrproject-rtos/zephyr/blob/ab9028518e1236213fad01ef3fe20b57b1699e0b/include/zephyr/drivers/usb/emul_bc12.h#L22).

Also add a reference to the sensor documentation [here](https://github.com/zephyrproject-rtos/zephyr/blob/ab9028518e1236213fad01ef3fe20b57b1699e0b/doc/hardware/peripherals/sensor.rst?plain=1#L230)

Reply:
> Also add a reference to the sensor documentation here

Would that just involve adding `.. doxygengroup:: my_new_group_name` below the current last line?

Reply:
> > Also add a reference to the sensor documentation here
> 
> Would that just involve adding `.. doxygengroup:: my_new_group_name` below the current last line?

Yes, that's correct.